hamocc/mo_param_bgc.F90

```diff
@@ -43,25 +43,199 @@ module mo_param_bgc
+  !********************************************************************
+  !     Radiation and attenuation parameters
+  !********************************************************************
+  ! parameters for sw-radiation attenuation
+  ! Analog to Moore et al., Deep-Sea Research II 49 (2002), 403-462
+  ! 1 kmolP = (122*12/60)*10^6 mg[Chlorophyl]
+  real, protected :: ctochl   = 60.        ! C to Chlorophyl ratio
+  real, protected :: atten_w  = 0.04       ! yellow substances attenuation in 1/m
+  real, protected :: atten_c              ! phytoplankton attenuation in 1/m
+  real, protected :: atten_uv = 0.33       !
+  real, protected :: atten_f  = 0.4        ! fraction of sw-radiation directly absorbed in surface layer
+                                           ! (only if FB_BGC_OCE) [feedback bgc-ocean]
+
+  !********************************************************************
+  !     Dust deposition and iron solubility parameters
+  !********************************************************************
+  !ik weight percent iron in dust deposition times Fe solubility
+  ! the latter three values come from Johnson et al., 1997
+  real, protected :: fetune     = 0.6          ! factor introduced to tune deposition/solubility
+  real, protected :: perc_diron
+  real, protected :: fesoly     = 0.5*1.e-9    ! max. diss. iron concentration in deep water
+  real, protected :: relaxfe    = 0.05/365.    ! 1/d complexation rate to relax iron concentration to fesoly
+
+  !********************************************************************
+  !     Phytoplankton parameters (incl. cyanobacteria)
+  !********************************************************************
+  real, protected :: phytomi = 1.e-11        !kmol/m3 - i.e. 1e-5 mmol P/m3 minimum concentration of phyto plankton (?js)
+  real, protected :: pi_alpha= 0.02*0.4      ! initial slope of production vs irradiance curve (alpha) (0.002 for 10 steps per day)
+  real, protected :: bkphy   = 4.e-8         !kmol/m3 - i.e. 0.04 mmol P/m3 half saturation constant
+  real, protected :: dyphy   = 0.004         !1/d -mortality rate of phytoplankton
+
+  ! N2-Fixation following the parameterization in Kriest and Oschlies, 2015.
+  ! Factors tf2, tf1 and tf0 are a polynomial (2nd order)
+  ! approximation to the functional relationship by Breitbarth et al. (2007),
+  ! for temperature dependence of Trichodesmium growth, their eq. (2)
+  ! The relation will be scaled to their max. growth rate, tff.
+  ! Note that the second order approx. is basically similar to their
+  ! function 2 for T-dependent nitrogen fixation multiplied by 4
+  ! (2 [N atoms per mole] * 12 [light hrs per day]/6 [C-atoms per N-atoms])
+  real, protected :: bluefix =  0.005         !1/d  ! nitrogen fixation rate by blue green algae (cyanobacteria)
+  real, protected :: tf2     = -0.0042
+  real, protected :: tf1     =  0.2253
+  real, protected :: tf0     = -2.7819
+  real, protected :: tff     =  0.2395
```

Comment:
I would tend to have tf* defined as parameters - these come from a specific parameterization, so I wouldn't think there is a need to have them as tunable parameters? (Anyway, this is obsolete with the new N-cycle, I guess)

Reply:
Well, they are still used - also with the N-cycle. Since there might be the wish to run `bluefix` only in the euphotic zone with the new hybrid coordinates, there might be the wish to tune them (while I suspect, we won't). Hence, I would be a bit hesitant to set these parameters a `parameter`.